torch_xla/experimental/tpu.py

```diff
@@ -94,18 +109,20 @@ def version() -> int:
   except requests.HTTPError as e:
     raise EnvironmentError('Failed to get TPU metadata') from e
 
-  match = re.match(r'^v(\d)-(\d+)$', env['ACCELERATOR_TYPE'])
+  match = re.match(r'^v(\d)-(\d+)$', env[xenv.ACCELERATOR_TYPE])
   return int(match.groups()[0])
 
 
 def get_worker_ips() -> List[str]:
   """Returns ordered list of TPU worker IPs from TPU metadata."""
-  metadata = _get_metadata('worker-network-endpoints')
-
-  # Workers have format 'hostname:uid:ip,hostname:uid:ip,...'
-  workers = metadata.split(',')
-  ips = [worker.split(':')[2] for worker in workers]
-
+  metadata = xu.getenv_as(xenv.TPU_WORKER_HOSTNAMES, str, '')
```

Comment:
nit: call this something more specific like `hostnames_env` (feel free to come up with a better name). I don't want readers to confuse this with GCE metadata, and we want to make it clearer to the reader this has a different format than the other usage of `metadata` down below.

Reply:
Done